### PR TITLE
Add Hummingbird theme in composer.json

### LINKED ISSUE #36182: Integrate an alpha release of the theme in the project
This is the step to integrate an alpha version in the core, the theme will be accessible but not used as the default one

- [x] Release the theme with the latest changes into a 0.2.0 version (see this PR https://github.com/PrestaShop/hummingbird/pull/600 - the version needs to be bumped to 0.2.0)
- [ ] Prepare a page in devdocs that explains the technical constraints for this theme (can be empty at first but we need the URL for the following task) The page can later be enriched -> moved to https://github.com/PrestaShop/PrestaShop/issues/36145
  - [ ] Explain that the theme is based on bootstrap5 and won't be compatible with all modules until they handle the theme -> moved to https://github.com/PrestaShop/PrestaShop/issues/36145
  - [ ] Explain the trick about integrating the Bootstrap Compatibility Layer -> moved to https://github.com/PrestaShop/PrestaShop/issues/36145
- [ ] Integrate hummingbird 0.2.0 into composer, adapt UI tests that don't need to install the theme anymore, integrate a warning popup when the theme is enabled to warn about bootstrap with a link to devdoc for more details (also see this issue https://github.com/PrestaShop/hummingbird/issues/581) -> moved to https://github.com/PrestaShop/PrestaShop/issues/36145

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add Hummingbird theme in composer.json and adapt UI tests
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | UI tests 🟢 
| UI Tests          | https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9796646790
| Sponsor company   | PrestaShop SA

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
         "prestashop/graphnvd3": "^2",
         "prestashop/gridhtml": "^2",
         "prestashop/gsitemap": "^4",
+        "prestashop/hummingbird": "dev-develop",
```
Comment (jolelievre) [inline composer.json:72]:
```suggestion
        "prestashop/hummingbird": "~0.2",
```

Comment (nicosomb) [reply composer.json:72]: @jolelievre for the classic theme, on develop branch, we put `dev-develop`. We only put tags on the version branches.

File: tests/UI/commonTests/BO/design/hummingbird.ts [hunk 2]
```diff
@@ -52,27 +45,6 @@ function installHummingbird(baseContext: string = 'commonTests-installHummingbir
       expect(numThemes).to.eq(1);
```
Comment (nesrineabdmouleh) [inline tests/UI/commonTests/BO/design/hummingbird.ts]:
```suggestion
      expect(numThemes).to.eq(2);
```

File: tests/UI/commonTests/BO/design/hummingbird.ts [hunk 3]
```diff
@@ -82,7 +54,7 @@ function installHummingbird(baseContext: string = 'commonTests-installHummingbir
   });
 }
 
-function uninstallHummingbird(baseContext: string = 'commonTests-uninstallHummingbird'): void {
+function disableHummingbird(baseContext: string = 'commonTests-disableHummingbird'): void {
   describe('Uninstall Hummingbird theme', async () => {
```
Comment (nesrineabdmouleh) [inline tests/UI/commonTests/BO/design/hummingbird.ts]:
```suggestion
  describe('Disable the Hummingbird theme', async () => {
```

File: composer.json (hunk 1 above)
Comment (jolelievre) [reply composer.json:72]: You're right it makes sense!

**Review by jolelievre — DISMISSED** [review]
Thanks @nicosomb

**Comment by nicosomb** [comment]
UI tests are failing, due to Hummingbird build steps. 

This PR is blocked, we have to improve the Hummingbird build (remove the `.env` step, etc.), release a new HB version and relaunch tests on this PR.

**Comment by nicosomb** [comment]
a quick test with my last commit https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9176558012

**Comment by nicosomb** [comment]
UI test https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9352955630

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: composer.lock [hunk 4]
```diff
@@ -4,20 +4,20 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "e9ed96112baedf6d3f99dd4d2f1bbbbb",
+    "content-hash": "241293f1538a544041bb741bf7fc93b1",
     "packages": [
         {
             "name": "api-platform/core",
-            "version": "v3.2.13",
+            "version": "v3.3.5",
```
Comment (jolelievre) [inline composer.lock]: You need to revert completely the changes on the composer.lock

And then run only `composer update prestashop/hummingbird`

**Review by jolelievre — DISMISSED** [review]

File: composer.json [hunk 5]
```diff
@@ -69,6 +69,7 @@
         "prestashop/graphnvd3": "^2",
         "prestashop/gridhtml": "^2",
         "prestashop/gsitemap": "^4",
+        "prestashop/hummingbird": "dev-develop#76f0070a140de899029036ad645aa08594a480d0",
```
Comment (jolelievre) [inline composer.json]:
```suggestion
        "prestashop/hummingbird": "dev-develop",
```
You don't need to specify the commit here, only the branch is needed The commit is then saved in the lock file every time we run `composer update` again

**Review by jolelievre — COMMENTED** [review]
Last modif

File: composer.json (hunk 5 above)
Comment (jolelievre) [reply composer.json]: BTW even the CI doesn't like it https://github.com/PrestaShop/PrestaShop/actions/runs/9496040516/job/26169578438?pr=36198 😅

Comment (nicosomb) [reply composer.json]: I know that but I want to launch CI with this pending PR https://github.com/PrestaShop/hummingbird/pull/629

**Comment by jolelievre** [comment]
Just curious about how long the hummingbird campaign take now ^^ https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/9496425478

Previously I believe these two were the longest jobs:
- campaign 08-12: 39min
- campaign 01-03: 25min

**Comment by nicosomb** [comment]
UI tests https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9657564198

**Comment by nicosomb** [comment]
UI tests with HB assets build in ga.tests.ui https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9662510503

**Comment by nesrineabdmouleh** [comment]
PR to fix a selector in the test library https://github.com/PrestaShop/ui-testing-library/pull/70 to merge at the same time

Ui tests => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/9713503812

**Review by Progi1984 — DISMISSED** [review]
Ok for UI Tests

**Comment by nicosomb** [comment]
@Progi1984 ga.tests.ui are green on your side?

**Review by Progi1984 — DISMISSED** [review]
Ok for UI Tests

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by nicosomb** [comment]
UI tests: https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9796646790

**Review by boherm — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]
Approved for UI Tests

**Review by florine2623 — DISMISSED** [review]
Hello @nicosomb ,

PR is well installed, hummingbird theme can be installed/uninstalled without issue.
Tests are green so it is QA ✅ 

https://github.com/PrestaShop/PrestaShop/assets/16019289/9d31551d-a6d0-4ab1-850f-01dbc464f2b1

Thanks!

**Comment by nicosomb** [comment]
UI tests are green 🎉

**Comment by nicosomb** [comment]
New UI tests after @nesrineabdmouleh commit https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/9804149210

**Review by boherm — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]
Ok for UI Tests

**Review by matks — APPROVED** [review]

**Pull request merged**